HomeWork1/Maksim_Panibrat_HW1_Vector.hpp

```diff
@@ -0,0 +1,320 @@
+#pragma once
+
+// iostream used only in analize and prints, can be deleted when program will be complited
+#include <iostream>
+
+template <typename T>
+class Vector {
+public:
+	Vector() = default;
+	Vector(size_t new_size) : capacity{ new_size }, size{ capacity } {
+		data = alloc_memory(capacity);
+		new(data)T[capacity]{};
+	}
+	Vector(const size_t new_size, const T value) : capacity{ new_size }, size{ capacity } {
+		data = alloc_memory(capacity);
+		new(data)T[capacity];
+
+		std::fill_n(data, capacity, value);
+	}
+	Vector(const std::initializer_list<T>& list) : Vector(list.size()) {
+		memory_copy(data, list.begin(), list.size());
+		size = list.size();
+	}
+	Vector(const T* rhs_begin, const T* rhs_end) {
+		capacity = std::distance(rhs_begin, rhs_end);
+		size = capacity;
+		data = alloc_memory(capacity);
+		new(data)T[capacity];
```

Comment:
```suggestion
	Vector(const T* rhs_begin, const T* rhs_end) : Vector(std::distance(rhs_begin, rhs_end)) {
```
Постарайтесь в своем коде переиспользовать уже существующий функционал кода.
его очень много